File: revision/intersection_two_arrays2.cpp
#include<iostream>
#include<vector>
#include<unordered_map>
using namespace std;
void input(vector<int>&nums){
   for(auto &i : nums){
       cin >> i;
   }
}
void display(vector<int>nums){
    for(auto const i : nums){
        cout << i << ' ' ;
    }
    cout << endl;
}
vector<int>Intersection(vector<int>& nums1, vector<int>& nums2) {
    unordered_map<int, int>f1, f2;
    vector<int>res;
    for(auto i: nums1){
        if(f1.find(i) == f1.end()){
            f1.insert(make_pair(i,1));
        } else{
            f1[i]++ ;
        }
    }
    for(auto i: nums2){
        if(f2.find(i) == f2.end()){
            f2.insert(make_pair(i,1));
        } else{
            f2[i]++ ;
        }
    }
    for(auto i : f1){
        if (f2.find(i.first) != f2.end()){
            int s = min(f1[i.first], f2[i.first]);
            while (s--){
                res.push_back(i.first);
            }
        }
    }
    return res;
}
int main()
{
    int m,n;
    cin >> m >> n;
    vector<int>nums1(m), nums2(n);
    input(nums1);
    input(nums2);
    display(Intersection(nums1, nums2));
    
    return 0;
}

File: revision/intersection_two_arrays2.py
from typing import List
def intersect(nums1:List[int], nums2:List[int])->List[int]:
    freq1 , freq2, res = dict(), dict(), []
    for x in nums1:
        freq1[x]  = freq1.get(x, 0) + 1
    for y in nums2:
        freq2[y]  = freq2.get(y, 0) + 1
    print(freq1, freq2)
    for k , v in freq1.items():
        if k in freq2:
            s = min(freq1.get(k), freq2.get(k))
            for i in range(s):
                res.append(k)
    return res
nums1 = list(map(int, input().split()))
nums2 = list(map(int, input().split()))
print(intersect(nums1, nums2))